### PR TITLE
fix(exec): set default ref on pull_request event

### PR DESCRIPTION
**Pull request by JordanSussman**
closes https://github.com/go-vela/community/issues/305

### CODE REVIEW

**Review by jbrockopp — APPROVED** [review]
LGTM

File: action/pipeline/exec.go [hunk 1]
```diff
@@ -46,7 +46,11 @@ func (c *Config) Exec(client compiler.Engine) error {
 	b.SetBranch(c.Branch)
 	b.SetDeploy(c.Target)
 	b.SetEvent(c.Event)
-	b.SetRef(c.Tag)
+	if c.Tag == "" && c.Event == constants.EventPull {
+		b.SetRef("refs/pull/1")
+	} else {
+		b.SetRef(c.Tag)
+	}
```
Comment (wass3r) [inline action/pipeline/exec.go:53]: wouldn't this set the ref to `c.Tag` when `c.Event != constants.EventPull`?

Comment (JordanSussman) [reply action/pipeline/exec.go:53]: Yes. Are you thinking that we should automatically set the ref for other event types if `c.Tag` is not defined?

Comment (wass3r) [reply action/pipeline/exec.go:53]: I think we're good. I didn't see that we were setting `c.Tag` previously. Just seems odd that we use `c.Tag` on all but Pull Requests.. sorry for the noise :)

**Review by wass3r — APPROVED** [review]

**Pull request merged**